src/main/java/org/craftercms/studio/impl/v2/repository/GitContentRepository.java

```diff
@@ -964,8 +964,12 @@ public void publish(String site, String sandboxBranch, List<DeploymentItemTO> de
                         commitId = deploymentItem.getCommitId();
                         path = helper.getGitPath(deploymentItem.getPath());
                         if (Objects.isNull(commitId)) {
-                            logger.warn("Skipping file " + path + " because commit id is null");
-                            continue;
+                            if (contentExists(site, path)) {
+                                commitId = getRepoLastCommitId(site);
```

Comment:
Will this guarantee the file being present in the index and a successful publish?

Reply:
yes